Update docs for cluster networks (#114)
harvester/docs: docs/install/harvester-configuration.md

  http-proxy: '{"httpProxy": "http://my.proxy", "httpsProxy": "https://my.proxy", "noProxy": "some.internal.svc"}'
  ui-source: auto
```

### `cluster_networks`

#### Definition

You can setup the default network in Harvester by configuring `cluster_networks`

> !!! note
>     Setup default cluster networks only works when Harvester is installed in "create" mode.
>     If you install Harvester in "join" mode, this setting is ignored.
>     Installing in "join" mode will adopt the cluster networks from the existing Harvester system.

Reviewer: ```suggestion
!!! note
    To configure the `cluster_networks`, Harvester needs to be installed in "create" mode.
    If you install Harvester in "join" mode, this setting is ignored.
    Installing in "join" mode will apply the `cluster_networks` configuration from the existing Harvester system.
```

#### Example

> The example setups the default physical NIC name of the VLAN network.

Reviewer: ```suggestion
The following example sets the default physical NIC name of the VLAN network.
```

  http-proxy: '{"httpProxy": "http://my.proxy", "httpsProxy": "https://my.proxy", "noProxy": "some.internal.svc"}'
  ui-source: auto
```

### `cluster_networks`

> 

Reviewer 2: add `_Available as of v1.0.1_`.

Author: added in L515

#### Definition

You can setup the default network in Harvester by configuring `cluster_networks`. Network configuration reference:

- `vlan`: Setup for VLAN network. The following fields are supported:
    - `enable`: enable VLAN network settings or not. Default value: `false`.
    - `description`: Additional information for `ClusterNetworks`. Default value: "".
    - `config`: `ClusterNetworks` configuration to be used. Valid configuration fields are:

>         - `defaultPhysicalNIC` (string, required): assign a physical NIC to be external entry of VLAN network

Reviewer: ```suggestion
        - `defaultPhysicalNIC` (string, required): assign a physical NIC to be external entry of VLAN network.
```

> - `vlan`: Setup for VLAN network. The following fields are supported by VLAN network:
>     - `enable` (bool, optional): enable VLAN network settings or not. Defaults to false.
>     - `description` (string, optional): Describe any comment or related information for ClusterNetworks. Defaults to "".
>     - `config` (dictionary, required): Specific type of `ClusterNetworks` configuration to be used. Valid configuration fields are:
>         - `defaultPhysicalNIC` (string, required): assign physical NIC to be external entry of VLAN network

Reviewer: ```suggestion
- `vlan`: Setup for VLAN network. The following fields are supported:
    - `enable`: enable VLAN network settings or not. Default value: `false`.
    - `description`: Additional information for `ClusterNetworks`. Default value: "".
    - `config`: `ClusterNetworks` configuration to be used. Valid configuration fields are:
        - `defaultPhysicalNIC` (string, required): assign a physical NIC to be external entry of VLAN network
```

Reviewer: The original settings list makes a lot of sense with the type and mandatory/optional value. However as the whole page does not have anything like this, it's breaks the "style".

However, I will check with the team and come with an updated version were we do have these information for all fields.

!!! note
    To configure the `cluster_networks`, Harvester needs to be installed in "create" mode.
    If you install Harvester in "join" mode, this setting is ignored.
    Installing in "join" mode will apply the `cluster_networks` configuration from the existing Harvester system.

#### Example

> The following example sets the default physical NIC name of the VLAN network.

Reviewer: ```suggestion
The following example sets the default physical NIC name of the VLAN network:
```
